src/functionsDeployHelper.ts

```diff
@@ -161,10 +184,124 @@ export function getFunctionsInfo(parsedTriggers: CloudFunctionTrigger[], project
         "functions",
         trigger.name,
       ].join("/");
-      functionsInfo.push(triggerDeepCopy);
+      if (!_.get(regionMap, region)) {
+        regionMap[region] = [];
+      }
+      regionMap[region].push(triggerDeepCopy);
     });
   });
-  return functionsInfo;
+  return regionMap;
+}
+
+/**
+ * Helper method to turn a RegionMap into a flat list of all functions in a deployment.
+ * @param regionMap A RegionMap for the deployment.
+ */
+export function flattenRegionMap(regionMap: RegionMap): CloudFunctionTrigger[] {
+  return _.chain(regionMap)
+    .map((value: CloudFunctionTrigger[]) => {
+      return value;
+    })
+    .flatten()
+    .value();
```

Comment:
What is your feeling about `_.chain`? Personally, I'm not a fan.

Is this equivalent?

```typescript
const triggers: CloudFunctionTrigger[] = [];
for (const [k, v] of Object.entries(regionMap)) {
  triggers.push(...v);
}
return triggers;
```

If you're really wanting to keep `_.chain` though, here's a simplification you can make:

```suggestion
  return _.chain(regionMap)
    .map((v: CloudFunctionTrigger[]) => v)
    .flatten()
    .value();
```

Reply:
I'm not the biggest fan of `_.chain` either - its tough to understand what going on with it if you're not familiar with lodash, and its always been unintuitive to me that it doesn't execute any of the chained functions until you call '.value()'. 

 However, as I'm working on the follow up PR, I've realized that I can get rid of this method entirely with some minor refactoring of checkIam.ts. For now, I think I'll go with the simplified _.chain approach - in a follow up PR, this should be gone entirely.

Reply:
heh - seriously though, my goal is to get rid of lodash entirely. So much of it can be done with plain JS (or a simple helper function), so it's not unattainable...

Reply:
+💯  to getting rid of lodash (especially the more esoteric bits) everywhere we're rewriting code.